x-pack/test/functional_endpoint/apps/endpoint/resolver.ts

```diff
@@ -0,0 +1,266 @@
+      const toTime = 'Now';
+      await esArchiver.load('endpoint/resolver_tree/api_feature');
+      await pageObjects.common.navigateToUrlWithBrowserHistory('endpoint', '/alerts');
+      // await pageObjects.common.sleep(4000);
+      await retry.try(async function() {
+        await testSubjects.existOrFail('superDatePickerShowDatesButton');
+      });
+      await pageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+      await testSubjects.existOrFail('alertListPage');
+      await (await testSubjects.find('alertTypeCellLink')).click();
+      await testSubjects.existOrFail('alertDetailFlyout');
+      await (await testSubjects.find('overviewResolverTab')).click();
+      await pageObjects.endpointAlerts.waitForTableToHaveData('resolverEmbeddable');
+    });
+
+    it('resolver column Process Name exits', async () => {
+      await testSubjects.existOrFail('tableHeaderCell_name_0');
+    });
+
+    it('resolver column Timestamp exits', async () => {
+      await testSubjects.existOrFail('tableHeaderCell_timestamp_1');
+    });
+
+    it('resolver Table and Node data same length', async () => {
+      let count = 1;
+      const tableData = await pageObjects.endpointAlerts.getEndpointAlertResolverTableData(
+        'resolverEmbeddable',
+        'tr'
+      );
+      await retry.try(async function() {
+        await (await testSubjects.find('zoom-out')).click();
+        const Nodes = await testSubjects.findAll('resolverNode');
+        expect(tableData.length - 1).to.eql(Nodes.length);
+        count++;
+      });
+
+      for (let i = 0; i < count; i++) {
+        await (await testSubjects.find('zoom-in')).click();
+      }
+    });
+
+    it('compare resolver Nodes and Table data', async () => {
+      const $: string[] = [];
+      let count = 1;
+      const tableData = await pageObjects.endpointAlerts.getEndpointAlertResolverTableData(
+        'resolverEmbeddable',
+        'tr'
+      );
+
+      await retry.try(async function() {
+        await (await testSubjects.find('zoom-out')).click();
+        const Nodes = await testSubjects.findAll('resolverNode');
+        expect(tableData.length - 1).to.eql(Nodes.length);
+        for (const value of Nodes) {
+          $.push(await value._webElement.getText());
+        }
+        count++;
+      });
+
+      for (let i = 0; i < $.length; i++) {
+        expect(tableData[i + 1][0]).to.eql($[i].split('\n')[1]);
+      }
+      for (let i = 0; i < count; i++) {
+        await (await testSubjects.find('zoom-in')).click();
+      }
+    });
+
+    it('resolver Nodes navigation Up', async () => {
+      const OriginalNodeDataStyle = await pageObjects.endpointAlerts.parseStyles(
+        'resolverNode',
+        'style'
+      );
+      await (await testSubjects.find('north-button')).click();
+
+      const NewNodeDataStyle = await pageObjects.endpointAlerts.parseStyles(
+        'resolverNode',
+        'style'
+      );
+      for (let i = 0; i < OriginalNodeDataStyle.length; i++) {
```

Comment:
during the meeting we discussed adding in validation that, for example, when you click the 'up' north button, that the 'left' value stays the same.  probably easy validation to add once you have all the data like here.  

another thought I had, with the 'up' button example:  we could assess the delta change for each resolver node for the 'top' value is the same, as opposed to just being greater/less than it was.  Meaning - the delta position for all nodes should be the same, so you can capture the value, assume it is right (if its greater /less than it should be, and walk the list comparing that all values are equally moved.  this could work for all of up / down / left / right.

Reply:
you can certainly add that in a separate pr if you wanted, this is fabulous work, igor.